owltorrent-core/src/main/java/edu/rice/owltorrent/core/serialization/SHA1Encryptor.java

```diff
@@ -0,0 +1,35 @@
+package edu.rice.owltorrent.core.serialization;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+
+/** SHA1 Encryptor */
+public class SHA1Encryptor {
+
+  /**
+   * Encrypt the input string using SHA1 algorithm
+   *
+   * @return the raw 20-byte SHA1 hash
+   */
+  public static byte[] encrypt(String input) {
+    try {
+      MessageDigest md = MessageDigest.getInstance("SHA-1");
+
+      return md.digest(input.getBytes());
+    } catch (NoSuchAlgorithmException e) {
+      throw new RuntimeException(e);
```

Comment:
this can be IllegalStateException instead. Not a necessary change tho

Reply:
https://docs.oracle.com/javase/8/docs/api/java/security/MessageDigest.html#getInstance-java.lang.String-
By definition, `public static MessageDigest getInstance(String algorithm) throws NoSuchAlgorithmException`